test/membrane_http_adaptive_stream/integration_test/sink_bin_integration_test.exs

```diff
@@ -0,0 +1,189 @@
+defmodule Membrane.HTTPAdaptiveStream.SinkBinIntegrationTest do
+  use ExUnit.Case, async: true
+
+  import Membrane.Testing.Assertions
+  alias Membrane.Testing
+
+  @doc """
+  The boolean flag below controls whether reference HLS content in fixtures directory will be created simultaneously with test content.
+  It should be set only when developing new HLS features that are expected to introduce changes to reference HLS files. Nevertheless it should
+  be done only locally to create and push new reference HLS files and this flag must not be set in remote repository. There is unit test in code below
+  that will cause CI to fail if this flag happens to be set on remote repository.
```

Comment:
Please add the note that when the new reference playlist was generated one should verify it manually if the playback is correct, for example using ffplay or other hls player.